src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -211,6 +211,20 @@ public GlobalSearchBar(JabRefFrame frame, StateManager stateManager, Preferences
 
         this.stateManager.activeSearchQueryProperty().addListener((obs, oldvalue, newValue) -> newValue.ifPresent(this::updateSearchResultsForQuery));
         this.stateManager.activeDatabaseProperty().addListener((obs, oldValue, newValue) -> stateManager.activeSearchQueryProperty().get().ifPresent(this::updateSearchResultsForQuery));
+        /*
+         * Attach a change listener on the focus property of searchField node.
+         * Every time the listener tracks a change on focus property value,
+         * (i.e. from active (user types query) to inactive / lost (user selects
+         * entry / completes search)), ensures the focus is lost
+         * (i.e. newValue : false), checks whether the search query is
+         * empty or not and finally records it into the search history
+         * through addSearchHistory method call.
```

Comment:
I would use line breaks to guide the reader. Moreover, the sentence is really too long to understand.

The text should not appear as generated one-by-one by reading the code.

Shorter version:

```suggestion
         * The listener tracks a change on the focus property value.
         * This happens, from active (user types a query) to inactive / focus lost (e.g., user selects an entry or triggers the search).
         * The search history should only be filled, if focus is lost.
```